src/allmydata/web/storage.py

```diff
@@ -127,55 +145,59 @@ def format_crawler_progress(self, p):
             return ["Next crawl in %s" % abbreviate_time(soon),
                     cycletime_s]
 
-    def render_lease_expiration_enabled(self, ctx, data):
+    @renderer
+    def storage_running(self, req, tag):
+        if self.storage:
+            return tag
+        return tag("No Storage Server Running")
+
+    @renderer
+    def lease_expiration_enabled(self, req, tag):
+        if not self.storage:
+            return tag("No storage server running.")
         lc = self.storage.lease_checker
         if lc.expiration_enabled:
-            return ctx.tag["Enabled: expired leases will be removed"]
+            return tag("Enabled: expired leases will be removed")
         else:
-            return ctx.tag["Disabled: scan-only mode, no leases will be removed"]
+            return tag("Disabled: scan-only mode, no leases will be removed")
 
-    def render_lease_expiration_mode(self, ctx, data):
+    @renderer
+    def lease_expiration_mode(self, req, tag):
+        if not self.storage:
+            return tag("No storage server running.")
```

Comment:
It occurs to me that all of these storage checks could be eliminated if the code instantiating this Element just never did so when there is no storage server (it could instantiate a plain Element with a different static template instead and just be done - maybe).

This isn't really _directly_ pertinent to this PR but it might be an interesting refactoring later on so I thought I'd write it down (though I should be writing in the issue tracker instead...).

Reply:
They are actually gone now!  You are right that they could be eliminated, but they were also directly pertinent to this PR.  Because I messed up initially.

The version prior to my edits rendered a big honking `"<h1>no storage server running</h1>"`, and nothing else.  My initial changes rendered the rest of the page too, with all the useless information, but I have changed it back to how it was in 561cd07.